tiledb/sm/filesystem/test/unit_uri.cc

```diff
@@ -224,6 +224,29 @@ TEST_CASE("URI: Test REST components", "[uri]") {
   CHECK(!URI("tiledb:///").get_rest_components(&ns, &array).ok());
 }
 
+TEST_CASE("URI: Test get_fragment_name", "[uri][get_fragment_name]") {
+  std::vector<std::pair<URI, std::optional<URI>>> cases = {
+      {URI("a randomish string"), std::nullopt},
```

Comment:
FYI. We have a global `using std::nullopt` in the common headers to avoid qualifying this.